setup.py

```diff
@@ -82,36 +85,105 @@ def process_gperf_file(gperf_file, output_file):
             version = m.group(1)
 assert version
 
+def build_extension(option):
+    sources = ['snudown.c', 'src/bufprintf.c']
+    sources.extend(c_files_in('html/'))
+    libraries = []
+    library_dirs= []
+    extra_compile_args = []
+    extra_link_args = []
+    extra_objects=[]
+
+    if option == 'translate':
+        library_dirs.append('translator-build')
+        libraries.append('snudownrust')
+
+    if 'rustcheck' in option:
```

Comment:
Since you're building these strings yourself, maybe instead of the `option` parameter have a `fake` bool and a `kind` value which picks the extension kind to build, e.g., `TRANSLATE`, `RUST_XCHECK` or `CLANG_XCHECK`. In other words, using a string to pick which extension to build and how to do it isn't very clean.